core/trino-spi/src/main/java/io/trino/spi/type/DecimalAggregationAccumulatorType.java

```diff
@@ -0,0 +1,131 @@
+                        singletonList(TypeSignatureParameter.numericParameter((long) numberOfNoOverflowValues))),
+                Slice.class);
+
+        if (numberOfNoOverflowValues < 0) {
+            throw new IllegalArgumentException("Invalid DecimalAggregationAccumulatorType numberOfNoOverflowValues " + numberOfNoOverflowValues);
+        }
+        this.numberOfNoOverflowValues = numberOfNoOverflowValues;
+    }
+
+    private static Type createDecimalAggregationAccumulatorType(int numberOfNoOverflowValues)
+    {
+        return new DecimalAggregationAccumulatorType(numberOfNoOverflowValues);
+    }
+
+    @Override
+    public int getFixedSize()
+    {
+        return numberOfNoOverflowValues * Long.BYTES + Long.BYTES;
+    }
+
+    @Override
+    public BlockBuilder createBlockBuilder(BlockBuilderStatus blockBuilderStatus, int expectedEntries, int expectedBytesPerEntry)
+    {
+        int maxBlockSizeInBytes;
+        if (blockBuilderStatus == null) {
+            maxBlockSizeInBytes = PageBuilderStatus.DEFAULT_MAX_PAGE_SIZE_IN_BYTES;
+        }
+        else {
+            maxBlockSizeInBytes = blockBuilderStatus.getMaxPageSizeInBytes();
+        }
+        return new DecimalAggregationAccumulatorBlockBuilder(
+                blockBuilderStatus,
+                Math.min(expectedEntries, maxBlockSizeInBytes / getFixedSize()), numberOfNoOverflowValues);
+    }
+
+    @Override
+    public BlockBuilder createBlockBuilder(BlockBuilderStatus blockBuilderStatus, int expectedEntries)
+    {
+        return createBlockBuilder(blockBuilderStatus, expectedEntries, getFixedSize());
+    }
+
+    @Override
+    public Object getObjectValue(ConnectorSession session, Block block, int position)
+    {
+        if (block.isNull(position)) {
+            return null;
+        }
+        return new SqlVarbinary(getSlice(block, position).getBytes());
+    }
+
+    @Override
+    public BlockBuilder createFixedSizeBlockBuilder(int positionCount)
+    {
+        return new DecimalAggregationAccumulatorBlockBuilder(null, positionCount, numberOfNoOverflowValues);
+    }
+
+    @Override
+    public void appendTo(Block block, int position, BlockBuilder blockBuilder)
+    {
+        if (block.isNull(position)) {
+            blockBuilder.appendNull();
+        }
+        else {
+            block.writePositionTo(position, blockBuilder);
+        }
+    }
+
+    @Override
+    public void writeSlice(BlockBuilder blockBuilder, Slice value)
+    {
+        writeSlice(blockBuilder, value, 0, value.length());
+    }
+
+    @Override
+    public void writeSlice(BlockBuilder blockBuilder, Slice value, int offset, int length)
+    {
+        if (length != getFixedSize()) {
+            throw new IllegalStateException("Expected entry size to be exactly " + getFixedSize() + " but was " + length);
+        }
+        for (int i = 0; i <= numberOfNoOverflowValues; i++) {
+            blockBuilder.writeLong(value.getLong(offset + i * Long.BYTES));
+        }
```

Comment:
nit: maybe writer overflow separately for sake of clean code. `<=` is easy to miss